Respect user's flags.
Don't define --enable-icc (that's what the CC env is for)
--- configure.ac.orig	2010-11-17 23:15:08.000000000 +0000
+++ configure.ac	2010-11-17 23:30:56.000000000 +0000
@@ -21,43 +21,14 @@
 
 # Include macros
 sinclude(acx_pthread.m4)
-sinclude(acx_prog_cc_optim.m4)
 
 # Display pakage and version number
 AC_MSG_RESULT([*********** Configuring:  $PACKAGE $VERSION ($date) **********])
 
-# Initialize the list of compilers to consider
-cclist="cc gcc"
-
-# Backup and reset the input CFLAGS and LDFLAGS
-mycflags="$CFLAGS"
-CFLAGS=""
-myldflags="$LDFLAGS"
-LDFLAGS=""
-
-# Provide special option for the Linux Intel C compiler
-AC_MSG_CHECKING([for Linux Intel C compiler mode])
-AC_ARG_ENABLE(icc,
-	[AC_HELP_STRING([--enable-icc],
-	[Enable special mode for compilation with the Intel compiler \
-(off by default)])],
-        use_icc="yes"
-        cclist="icc $cclist"
-	AC_MSG_RESULT([yes]),
-        use_icc="no"
-	AC_MSG_RESULT([no]))
-
-
-# Checks for programs.
-# GCC is chosen last because it is likely to yield less optimized code
-AC_LANG(C)
-AC_PROG_CC([$cclist])
+AC_PROG_CC
 # C Compiler: Check that it is ANSI C and POSIX-compliant
 AM_PROG_CC_STDC
 AC_ISC_POSIX
-ACX_PROG_CC_OPTIM
-AC_DISABLE_STATIC
-#AC_DISABLE_SHARED
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 
@@ -67,7 +38,7 @@
 # Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h sys/mman.h time.h unistd.h)
-if test $use_icc = "yes" -a $CC = "icc"; then
+if test $CC = "icc"; then
 AC_CHECK_HEADERS(mathimf.h)
 fi
 
